storage/series/series.go

```diff
@@ -445,6 +447,32 @@ func (s *dbSeries) mergeBlock(
 	return nil
 }
 
+func (s *dbSeries) Update(blk block.DatabaseBlock) error {
+	s.Lock()
+	defer s.Unlock()
+
+	// ensure series is bootstraped before allowing any updates
+	if s.bs != bootstrapped {
+		return errSeriesNotBootstrapped
+	}
+
+	// we only accept updates to non-buffered data
+	// ensure buffer is past provided time
+	min, _ := s.buffer.MinMax()
+	if !blk.StartTime().Before(min) {
+		return errSeriesUpdateBuffered
+	}
+
+	existingBlocks := s.blocks
+	err := s.mergeBlock(existingBlocks, blk)
+	if err != nil {
+		return err
+	}
+	s.blocks = existingBlocks
```

Comment:
Don't need to reset s.blocks back to existing blocks here, mergeBlock performs update in place.